modules/storages/db/migrate/20230321194150_add_project_folder_to_projects_storages.rb

```diff
@@ -0,0 +1,47 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class AddProjectFolderToProjectsStorages < ActiveRecord::Migration[7.0]
+  def up
+    create_enum :project_folder_modes, %w[inactive manual]
+
+    change_table :projects_storages do |table|
+      table.string :project_folder_id
+      table.enum :project_folder_mode, enum_type: :project_folder_modes, default: :inactive, null: false
+    end
+  end
```

Comment:
If I am not mistaken, an ENUM in ActiveRecord also defines a scope. So, it might make sense to add an index on it. For example, at some point we want to select all projects that have automagically managed folders. When having thousand projects, an index would become handy.